xinference/model/audio/core.py

```diff
@@ -150,17 +150,19 @@ def create_audio_model_instance(
     model_uid: str,
     model_name: str,
     download_hub: Optional[Literal["huggingface", "modelscope", "csghub"]] = None,
+    model_path: Optional[str] = None,
     **kwargs,
 ) -> Tuple[Union[WhisperModel, ChatTTSModel, CosyVoiceModel], AudioModelDescription]:
     model_spec = match_audio(model_name, download_hub)
-    model_path = cache(model_spec)
+    if not model_path:
+        model_path = cache(model_spec)
     model: Union[WhisperModel, ChatTTSModel, CosyVoiceModel]
     if model_spec.model_family == "whisper":
-        model = WhisperModel(model_uid, model_path, model_spec, **kwargs)
+        model = WhisperModel(model_uid, model_spec, model_path, **kwargs)
     elif model_spec.model_family == "ChatTTS":
-        model = ChatTTSModel(model_uid, model_path, model_spec, **kwargs)
+        model = ChatTTSModel(model_uid, model_spec, model_path, **kwargs)
```

Comment:
You skip the `device` parameter which leads to GPU CI failed, since you change the order of the parameters.

Reply:
I did these previously with the intention of fixing these errors: https://github.com/Valdanitooooo/inference/actions/runs/10120724485/job/27990814107
Now I change `if not model_path:` to `if model_path is None:` , Because the `model_path` has already been verified earlier
```
        if model_path is not None:
            if not os.path.exists(model_path):
                raise ValueError(
                    f"Invalid input. `model_path`: {model_path} File or directory does not exist."
                )
```